homeassistant/components/device_automation/condition.py

```diff
@@ -51,20 +52,38 @@ async def async_get_conditions(
         """List conditions."""
 
 
-async def async_validate_condition_config(
-    hass: HomeAssistant, config: ConfigType
-) -> ConfigType:
-    """Validate device condition config."""
-    return await async_validate_device_automation_config(
-        hass, config, cv.DEVICE_CONDITION_SCHEMA, DeviceAutomationType.CONDITION
-    )
+class DeviceCondition(Condition):
+    """Device condition."""
 
+    def __init__(self, hass: HomeAssistant, config: ConfigType) -> None:
+        """Initialize condition."""
+        self._config = config
+        self._hass = hass
 
-async def async_condition_from_config(
-    hass: HomeAssistant, config: ConfigType
-) -> condition.ConditionCheckerType:
-    """Test a device condition."""
-    platform = await async_get_device_automation_platform(
-        hass, config[CONF_DOMAIN], DeviceAutomationType.CONDITION
-    )
-    return trace_condition_function(platform.async_condition_from_config(hass, config))
+    @classmethod
+    async def async_validate_condition_config(
+        cls, hass: HomeAssistant, config: ConfigType
+    ) -> ConfigType:
+        """Validate device condition config."""
+        return await async_validate_device_automation_config(
+            hass, config, cv.DEVICE_CONDITION_SCHEMA, DeviceAutomationType.CONDITION
+        )
+
+    async def async_condition_from_config(self) -> condition.ConditionCheckerType:
+        """Test a device condition."""
+        platform = await async_get_device_automation_platform(
+            self._hass, self._config[CONF_DOMAIN], DeviceAutomationType.CONDITION
+        )
+        return trace_condition_function(
+            platform.async_condition_from_config(self._hass, self._config)
+        )
+
+
+CONDITIONS: dict[str, type[Condition]] = {
+    "device": DeviceCondition,
+}
+
+
+async def async_get_conditions(hass: HomeAssistant) -> dict[str, type[Condition]]:
+    """Return the sun conditions."""
+    return CONDITIONS
```

Comment:
```suggestion
async def async_get_conditions(hass: HomeAssistant) -> dict[str, type[Condition]]:
    """Return the sun conditions."""
    return {
	    "device": DeviceCondition,
	}
```

Is there any advantage in having a CONST for this? If not, it is simpler/cleaner to just return the dict directly (same for sun).

Not a big deal, but since this will become the "guide" for other condition platforms, I think it would be a nice improvement.